src/zarr/abc/store.py

```diff
@@ -386,6 +391,60 @@ async def _get_many(
         for req in requests:
             yield (req[0], await self.get(*req))
 
+    async def getsize(self, key: str) -> int:
+        """
+        Return the size, in bytes, of a value in a Store.
+
+        Parameters
+        ----------
+        key : str
+
+        Returns
+        -------
+        nbytes : int
+            The size of the value (in bytes).
+
+        Raises
+        ------
+        FileNotFoundError
+            When the given key does not exist in the store.
+        """
+        # Note to implementers: this default implementation is very inefficient since
+        # it requires reading the entire object. Many systems will have ways to get the
+        # size of an object without reading it.
+        value = await self.get(key, prototype=default_buffer_prototype())
+        if value is None:
+            raise FileNotFoundError(key)
+        return len(value)
+
+    async def getsize_prefix(self, prefix: str) -> int:
+        """
+        Return the size, in bytes, of all values under a prefix.
+
+        Parameters
+        ----------
+        prefix : str
+            The prefix of the directory to measure.
```

Comment:
Can we offer implementers the following in documentation?:

This function will be called by zarr using a `prefix` that is the path of a group, an array, or the root. Implementations can choose to do undefined behavior when that is not the case.

Reply:
Sure... I was hoping we could somehow ensure that we don't call it with anything other than a group / array / root path, but users can directly use `Store.getsize_prefix` and they can do whatever. 

LMK if you want any more specific guidance on what to do (e.g. raise a `ValueError`). I'm hesitant about trying to force required exceptions into an ABC / interface.

Reply:
> I'm hesitant about trying to force required exceptions into an ABC / interface.

And now I'm noticing that I've done exactly that in `getsize` :)